src/js/src/components/SearchBox.vue

```diff
@@ -94,14 +94,30 @@ export default {
       this.$router.history.current.query.grouping === 'true' ? this.updateSolrSettingGrouping(true) : null
       this.$router.history.current.query.imgSearch === 'true' ? this.updateSolrSettingImgSearch(true) : null
       this.$router.history.current.query.urlSearch === 'true' ? this.updateSolrSettingUrlSearch(true) : null
-      this.requestSearch({query:this.query, facets:this.searchAppliedFacets, options:this.solrSettings})
-      this.requestFacets({query:this.query, facets:this.searchAppliedFacets, options:this.solrSettings})
+      if(this.solrSettings.imgSearch === true) {
+          console.log('firing image search')
+           this.requestImageSearch({query:this.query})
+           this.$_pushSearchHistory('SolrWayback', this.query, this.searchAppliedFacets, this.solrSettings)
+          return
+        }
+        if(this.solrSettings.urlSearch === true) {
+          console.log('fireing url search')
+          return
+        }
```

Comment:
Consider using if, else-if and else pattern instead of if with return statements inside. Does the same in this case and easier to read.

Reply:
:+1: